Mathlib/Data/QPF/Multivariate/Constructions/Fix.lean

```diff
@@ -0,0 +1,370 @@
+
+theorem recF_eq' {α : TypeVec n} {β : Type _} (g : F (α.append1 β) → β) (x : q.P.W α) :
+    recF g x = g (abs (appendFun id (recF g) <$$> q.P.wDest' x)) := by
+  apply q.P.w_cases _ x
+  intro a f' f
+  rw [recF_eq, q.P.wDest'_wMk, MvPFunctor.map_eq, appendFun_comp_splitFun, TypeVec.id_comp]
+set_option linter.uppercaseLean3 false in
+#align mvqpf.recF_eq' MvQPF.recF_eq'
+
+/-- Equivalence relation on W-types that represent the same `Fix F`
+value -/
+inductive WEquiv {α : TypeVec n} : q.P.W α → q.P.W α → Prop
+  | ind (a : q.P.A) (f' : q.P.drop.B a ⟹ α) (f₀ f₁ : q.P.last.B a → q.P.W α) :
+    (∀ x, WEquiv (f₀ x) (f₁ x)) → WEquiv (q.P.wMk a f' f₀) (q.P.wMk a f' f₁)
+  | abs (a₀ : q.P.A) (f'₀ : q.P.drop.B a₀ ⟹ α) (f₀ : q.P.last.B a₀ → q.P.W α) (a₁ : q.P.A)
+    (f'₁ : q.P.drop.B a₁ ⟹ α) (f₁ : q.P.last.B a₁ → q.P.W α) :
+    abs ⟨a₀, q.P.appendContents f'₀ f₀⟩ = abs ⟨a₁, q.P.appendContents f'₁ f₁⟩ →
+      WEquiv (q.P.wMk a₀ f'₀ f₀) (q.P.wMk a₁ f'₁ f₁)
+  | trans (u v w : q.P.W α) : WEquiv u v → WEquiv v w → WEquiv u w
+set_option linter.uppercaseLean3 false in
+#align mvqpf.Wequiv MvQPF.WEquiv
+
+theorem recF_eq_of_wequiv (α : TypeVec n) {β : Type _} (u : F (α.append1 β) → β) (x y : q.P.W α) :
+    WEquiv x y → recF u x = recF u y := by
+  apply q.P.w_cases _ x
+  intro a₀ f'₀ f₀
+  apply q.P.w_cases _ y
+  intro a₁ f'₁ f₁
+  intro h
+  -- porting note: induction on h doesn't work.
+  refine' @WEquiv.recOn _ _ _ _ _ (λ a a' _ => recF u a = recF u a') _ _ h _ _ _
+  · intros a f' f₀ f₁ _h ih; simp only [recF_eq, Function.comp]
+    congr; funext; congr; funext; apply ih
+  · intros a₀ f'₀ f₀ a₁ f'₁ f₁ h; simp only [recF_eq', abs_map, MvPFunctor.wDest'_wMk, h]
+  · intros x y z _e₁ _e₂ ih₁ ih₂; exact Eq.trans ih₁ ih₂
+set_option linter.uppercaseLean3 false in
+#align mvqpf.recF_eq_of_Wequiv MvQPF.recF_eq_of_wequiv
+
+theorem WEquiv.abs' {α : TypeVec n} (x y : q.P.W α)
+                    (h : MvQPF.abs (q.P.wDest' x ) = MvQPF.abs (q.P.wDest' y)) :
+    WEquiv x y := by
+  revert h
+  apply q.P.w_cases _ x
+  intro a₀ f'₀ f₀
+  apply q.P.w_cases _ y
+  intro a₁ f'₁ f₁
+  apply WEquiv.abs
+set_option linter.uppercaseLean3 false in
+#align mvqpf.Wequiv.abs' MvQPF.WEquiv.abs'
+
+theorem WEquiv.refl {α : TypeVec n} (x : q.P.W α) : WEquiv x x := by
+  apply q.P.w_cases _ x; intro a f' f; exact WEquiv.abs a f' f a f' f rfl
+set_option linter.uppercaseLean3 false in
+#align mvqpf.Wequiv.refl MvQPF.WEquiv.refl
+
+theorem WEquiv.symm {α : TypeVec n} (x y : q.P.W α) : WEquiv x y → WEquiv y x := by
+  intro h; induction h
+  case ind a f' f₀ f₁ _h ih => exact WEquiv.ind _ _ _ _ ih
+  case abs a₀ f'₀ f₀ a₁ f'₁ f₁ h => exact WEquiv.abs _ _ _ _ _ _ h.symm
+  case trans x y z _e₁ _e₂ ih₁ ih₂ => exact MvQPF.WEquiv.trans _ _ _ ih₂ ih₁
+set_option linter.uppercaseLean3 false in
+#align mvqpf.Wequiv.symm MvQPF.WEquiv.symm
+
+/-- maps every element of the W type to a canonical representative -/
+def wrepr {α : TypeVec n} : q.P.W α → q.P.W α :=
```

Comment:
```suggestion
def wRepr {α : TypeVec n} : q.P.W α → q.P.W α :=
```
This is a minor suggestion, feel free to ignore it, but to it feels `wRepr` should be two words, so `Repr` capitalized. If you do make this change, also change it down below

Reply:
I think not b/c it's not a `Repr` in the sense of the typeclss